src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        const { extraData } = action;
+
+        currentState = { ...currentState, ...extraData };
+        break;
+
+      case 'removeProperties':
+        const { keysToRemove } = action;
+
+        for (const key of keysToRemove) {
+          if (currentState.hasOwnProperty(key)) {
+            delete currentState[key];
+          }
+        }
+
+        break;
+
+      case 'clear':
+        currentState = {};
+        break;
+    }
```

Comment:
While you have implemented the switch statement correctly, it is a good practice to include a default case for error handling. This ensures that your function can handle unexpected action types gracefully.